nemo/core/classes/modelPT.py

```diff
@@ -1177,6 +1177,7 @@ def set_trainer(self, trainer: Trainer):
         Args:
             trainer: PyTorch Lightning Trainer object.
         """
+        self.trainer = trainer
```

Comment:
Why do we keep both _trianer and trianer?

Reply:
PTL 1.2.8+ refers to self.trainer in order to get values such as self.node_rank, self.global_rank etc. So it needs to be explicitly updated. The `_trainer` is for backward compat with older NeMo models.

Reply:
In that case, why haven't we change our codes to use _trainer instead of trainer? It is just a find and replace?

Reply:
They both are same thing and are practically the same reference. Newer code should just use self.trainer directly.